sublime_linter.py

```diff
@@ -278,9 +278,10 @@ def input(self, *args):
         return SublimeLinterClearErrorsFilenameInput()
 
     def run(self, filename):
+        # type: (str) -> None
         """Clear errors for one or all filenames."""
         if filename == "<All>":
-            errors_to_clear = persist.file_errors
+            errors_to_clear = persist.file_errors  # type: Mapping[str, List[LintError]]
             persist.file_errors = defaultdict(list)
         else:
             errors_to_clear = {filename: persist.file_errors.pop(filename)}
```

Comment:
A defaultdict is not a dict so we basically downcast to something, here: `Mapping`. I think this is an mypy error. We only use `.items()` here so a sophisticated type checker should infer e.g. they state `Dict` but they only use `Mapping`. 

Just like when you do `int -> int -> int` with the implementation `a + b` the type system actually knows that the type can be `Summable[T] -> Summable[T]`. 

So, a `type ignore` would maybe be preferrable?

Reply:
I have no experience with mypy, but as `Mapping` is fixing the error it is probably better than a ignore which I guess could swallow other errors.